src/app/frontend/replicasetdetail/replicaset_service.js

```diff
@@ -64,4 +66,41 @@ export class DeleteReplicaSetService {
 
     return deferred.promise;
   }
+
+  /**
+   * Opens an update replica set dialog. Returns a promise that is resolved/rejected when user wants
+   * to update the replicas. Nothing happens when user clicks cancel on the dialog.
+   *
+   * @param {!backendApi.ReplicaSetDetail} replicaSetDetail
+   * @returns {!angular.$q.Promise}
+   */
+  showUpdateReplicasDialog(replicaSetDetail) {
+    return showUpdateReplicasDialog(
+        this.mdDialog_, replicaSetDetail, this.updateReplicas.bind(this));
+  }
+
+  /**
+   * Updates number of replicas in replica set.
+   *
+   * @param {number} replicasCount
+   * @param {!backendApi.ReplicaSetDetail} replicaSetDetail
+   * @param {!function(!backendApi.ReplicaSetSpec)=} opt_callback
+   * @param {!function(!angular.$http.Response)=} opt_errback
+   * @return {angular.$q.Promise}
+   */
+  updateReplicas(replicasCount, replicaSetDetail, opt_callback, opt_errback) {
+    let resource = getReplicaSetSpecPodsResource(
+        new StateParams(replicaSetDetail.namespace, replicaSetDetail.name), this.resource_);
+    let deferred = this.q_.defer();
+
+    /** @type {!backendApi.ReplicaSetSpec} */
+    let replicaSetSpec = {
+      replicas: replicasCount,
+    };
+
+    resource.save(replicaSetSpec, opt_callback, opt_errback);
+    // TODO(floreks): Think about refreshing data after update.
```

Comment:
Maybe not. I've left this because I still thought that on some point we will make 3(4)-way data binding in order to reflect changes in backend. After decreasing number of pods page is refreshed and we can see that desired number of pods is 1, but still in logs there are more pods and current number of them is higher.